ctapipe/image/reducer.py

```diff
@@ -42,5 +116,68 @@ class NullDataVolumeReducer(DataVolumeReducer):
     Perform no data volume reduction
     """
 
-    def __call__(self, waveforms):
-        return waveforms
+    def select_pixels(self, waveforms, telid=None, selected_gain_channel=None):
+        mask = waveforms != 0
+        return mask
+
+
+class TailCutsDataVolumeReducer(DataVolumeReducer):
+    """
+    Reduce the time integrated shower image in 3 Steps:
+
+    1) Select pixels with tailcuts_clean.
+    2) Add iteratively all pixels with Signal S >= boundary_thresh
+       with ctapipe module dilate until no new pixels were added.
+    3) Adding new pixels with dilate to get more conservative.
+    """
+    end_dilates = IntTelescopeParameter(
+        default_value=1,
+        help="Number of how many times to dilate at the end."
+    ).tag(config=True)
+    picture_thresh = FloatTelescopeParameter(
+        default_value=8.0,
+        help="Picture threshold for the first tailcuts_clean step. All "
+             "pixels above are selected."
+    ).tag(config=True)
+    boundary_thresh = FloatTelescopeParameter(
+        default_value=4.0,
+        help="Boundary threshold for the first tailcuts_clean step and "
+             "also for the second iteration step."
+    ).tag(config=True)
+    min_number_picture_neighbors = IntTelescopeParameter(
+        default_value=0,
+        help="A picture pixel survives tailcuts_clean only if it has at "
+             "least this number of picture neighbors. This has no effect "
+             "in case keep_isolated_pixels is True."
+    ).tag(config=True)
+
+    def select_pixels(self, waveforms, telid=None, selected_gain_channel=None):
+        camera_geom = self.subarray.tel[telid].camera.geometry
+        # Pulse-integrate waveforms
+        charge, _ = self.image_extractor(
+            waveforms, telid=telid, selected_gain_channel=selected_gain_channel
+        )
+
+        # 1) Step: TailcutCleaning at first
+        mask = tailcuts_clean(
+            geom=camera_geom,
+            image=charge,
+            picture_thresh=self.picture_thresh.tel[telid],
+            boundary_thresh=self.boundary_thresh.tel[telid],
+            min_number_picture_neighbors=self.min_number_picture_neighbors.tel[telid],
+            keep_isolated_pixels=False
```

Comment:
This is maybe not a good idea for a data volume reduction right?

We probably want to be conservative here, so not removing bright pixels even if they are isolated might be not a good idea.

Reply:
Ah yes you're right, i'll swap it to the `BoolTelescopeParaemter` now.